### PR TITLE
Persisting MetaMetrics opt-in screen pre account setup

### LINKED ISSUE #10675: Opting in/and out of MetaMetrics in the onboarding flow does not save before creating or importing an account
**Describe the bug**
Before you either import or a create an account, navigating back will re-prompt you to opt in or out of Metametrics. (in the case of opt-in, it is already so indicated in the app state.)

**Steps to reproduce (REQUIRED)**

1. Start MetaMask
2. Select "Import Wallet"
3. Decline or accept MetaMetrics
4. When at the import screen, hit the back button
5. Find yourself back at the Metametrics screen
6. Repeat steps 1 - 5 with the Create flow

**Expected behavior**
Opting in or out of MetaMetrics should persist after one interaction

### PR DESCRIPTION
**Pull request by ryanml**
Fixes MetaMask/metamask-extension#10675

Test plan defined in issue

### CODE REVIEW

File: app/scripts/controllers/metametrics.js [hunk 1]
```diff
@@ -132,7 +133,11 @@ export default class MetaMetricsController {
     } else if (participateInMetaMetrics === false) {
       metaMetricsId = null;
     }
-    this.store.updateState({ participateInMetaMetrics, metaMetricsId });
+    this.store.updateState({
+      participateInMetaMetrics,
+      metaMetricsId,
+      metaMetricsParticipationSet: true,
```
Comment (brad-decker) [inline app/scripts/controllers/metametrics.js:139]: elsewhere we check if 'participateInMetaMetrics' is null, which is its default value, and once a user selects one of the two choices it becomes true or false. I do like the consistency of having a literal value in state, and it's definitely more explicit. I'm wondering, though, if we could make it slightly more explicit with "metaMetricsOnboardingCompleted" ... as we *can* check if participateInMetaMetrics has been set by comparing it to true/false.

**Review by brad-decker — CHANGES_REQUESTED** [review]
one minor suggestion, good work!

**Comment by Gudahtt** [comment]
I'm not sure if this is something we want from a design perspective, because it makes it impossible for the user to change their mind during onboarding about metrics. e.g. if I accept metrics, then second-guess my choice and click back, I can't undo that decision without completing onboarding and navigating to the settings. By that time _many_ metric events will have been dispatched, which might have been undesired.

**Comment by brad-decker** [comment]
> I'm not sure if this is something we want from a design perspective, because it makes it impossible for the user to change their mind during onboarding about metrics. e.g. if I accept metrics, then second-guess my choice and click back, I can't undo that decision without completing onboarding and navigating to the settings. By that time many metric events will have been dispatched, which might have been undesired.

I think I agree. I'm not sure which case is more dominant:
1) User abandons onboarding and immediately restarts
2) User abandons onboarding and starts back up after some time

In the case of 1, I can see being presented with these options again being a slight annoyance. Meanwhile, in the case of 2, I can see being shown the opt-in again being good because I may have completely forgotten about this and what the implications were. 

If we move forward with this, we'd be solving a slight annoyance for 1 and possibly making a *worse* experience for 2. Imagine being a new MetaMask user, you get through part of onboarding and then stop to research more. They come back with confidence and become a user. Later, they discover through twitter, blog posts, etc that MetaMask has analytics and have to find it in settings.

**Comment by ryanml** [comment]
If we feel this results in a more harmful UX (Scenario 2) - totally open to closing this PR. Just regarding scenario 1, when I had hit "No Thanks" and backtracked, and was presented with the opt-in again, I felt as though it seemed, "Did the opt-out really take affect?"

**Comment by ryanml** [comment]
Potential alternative, perhaps only introduce this new behavior if the user chooses "No Thanks", and if they choose to Accept, still allow for the old backtracking behavior?

**Comment by Gudahtt** [comment]
That alternative would be better from a privacy perspective, but still maybe confusing from a usability perspective? And it might result in less users opting in.

>when I had hit "No Thanks" and backtracked, and was presented with the opt-in again, I felt as though it seemed, "Did the opt-out really take affect?"

I'm curious why you felt this way :thinking:. As a user I find it mildly inconvenient to have to choose again, but it'd never occur to me to think that seeing it again implied it didn't work the first time - not after pressing 'Back'.

Would it help if 'Back' didn't skip the opt-in page? Maybe that's partly why you got the sense it didn't work - because it didn't act like a "back" button typically would by bringing you back one step? Just a guess.

**Comment by ryanml** [comment]
I think ultimately given the sentiments expressed this both A) Does not solve any perceived issue B) Creates a worse experience. Closing this.

**Pull request closed without merging**